### PR TITLE
CQ-4345810 Screenreader Accessibility

### PR DESCRIPTION
**Pull request by barshat7**
## Description

Screen Reader Priority Accessibility

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/textinput.html [hunk 1]
```diff
@@ -27,27 +27,32 @@
...
-              aria-label="${text.label.visible && text.label.value}"
+              aria-label="${text.htmlScreenReaderText && text.screenReaderText != '$description' ? text.htmlScreenReaderText : text.label.value}"
+              aria-labelledby="${text.screenReaderText == '$description' ? longDescriptionId : ''}"
               class="cmp-adaptiveform-textinput__widget"
               name="${text.name}"
               readonly="${text.readOnly}"
               required="${text.required}"
               placeholder="${text.placeHolder}"
-              data-cmp-required-message="${text.constraintMessages.required}">${text.default}</textarea>
+              data-cmp-required-message="${text.constraintMessages.required}"
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/textinput.html:42]: pls remove **data-cmp-required-message**

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/textinput.html [hunk 2]
```diff
@@ -27,27 +27,32 @@
...
+           aria-label="${text.htmlScreenReaderText && text.screenReaderText != '$description' ? text.htmlScreenReaderText : text.label.value}"
+           aria-labelledby="${text.screenReaderText == '$description' ? longDescriptionId : ''}"
            type="text"
            name="${text.name}"
            value="${text.default}"
            readonly="${text.readOnly}"
            required="${text.required}"
            placeholder="${text.placeHolder}"
-           data-cmp-required-message="${text.constraintMessages.required}"/>
+           data-cmp-required-message="${text.constraintMessages.required}"
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/textinput.html:54]: pls remove data-cmp-required-message

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/textinput.html (hunk 1 above)
Comment (barshat7) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/textinput.html:42]: Also removed in PR https://github.com/adobe/aem-core-forms-components/pull/140 . Please review and merge https://github.com/adobe/aem-core-forms-components/pull/140 first and I will resolve the conflicts and reopen this PR.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/textinput.html [hunk 3]
```diff
@@ -27,27 +27,32 @@
...
               required="${text.required}"
               placeholder="${text.placeHolder}"
-              data-cmp-required-message="${text.constraintMessages.required}">${text.default}</textarea>
+              data-cmp-required-message="${text.constraintMessages.required}"
+              aria-describedby="${component.id}-errorMessage">${text.default}</textarea>
     <input data-sly-test="${!useTextarea}"
            class="cmp-adaptiveform-textinput__widget"
-           aria-label="${text.label.visible && text.label.value}"
+           aria-label="${text.htmlScreenReaderText && text.screenReaderText != '$description' ? text.htmlScreenReaderText : text.label.value}"
+           aria-labelledby="${text.screenReaderText == '$description' ? longDescriptionId : ''}"
```
Comment (nit23uec) [inline ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/textinput.html:47]: if  `aria-labelledby` is empty then label tag is ignored.
`aria-label` and` aria-labelledby`  should only be included when label is hidden and screenReaderText != label.

**Review by nit23uec — CHANGES_REQUESTED** [review]
comments inline.

File: ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/textinput.html (hunk 1 above)
Comment (vdua) [reply ui.apps/src/main/content/jcr_root/apps/core/fd/components/form/textinput/v1/textinput/textinput.html:42]: #193 was merged instead of #140

**Pull request closed without merging**